nmrfx-utils/src/main/java/org/nmrfx/utils/properties/ListOperationItem.java

```diff
@@ -35,38 +37,31 @@
 public class ListOperationItem extends OperationItem implements ObservableObjectValue<String> {
 
     ArrayList value;
-    ArrayList defaultValue;
+    ArrayList<String> defaultValue;
     ChangeListener<? super String> listener;
     /**
      * This enables us to see the type of Unit that we interpret the List as.
      */
     ChoiceOperationItem typeSelector;
 
-    ArrayList<String> listTypes;
-
     /**
      *
      * @param listener
      * @param defaultValue optional default value for the List.
-     * @param listTypes The types of objects that will be contained in the list.
-     * The List can only support one type of item at a time.
      * @param category
      * @param name
      * @param description
      * @param typeSelector
      */
-    public ListOperationItem(ChangeListener listener, ArrayList defaultValue, ArrayList<String> listTypes, String category, String name, String description, ChoiceOperationItem typeSelector) {
+    public ListOperationItem(ChangeListener<? super String> listener, List<String> defaultValue, String category, String name, String description, ChoiceOperationItem typeSelector) {
```

Comment:
Made these changes because sonar wanted a type specified and to use List over ArrayList

Reply:
`? super String` is strange, because String is final (meaning, there can be no subclasses).

Reply:
Yeah I wasn't sure about this one, sonar wanted a type so I used the type that was used a few lines above: https://github.com/nanalysis/nmrfx/blob/dd83badf98ccf0847d0c0d02a7c58661a78b1228/nmrfx-utils/src/main/java/org/nmrfx/utils/properties/ListOperationItem.java#L39
I think this one might actually be String and its super Object, according to this https://docs.oracle.com/javase/tutorial/java/generics/lowerBounded.html